e2e/pages/app/badge/badge.testpage.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component } from '@angular/core';
+import { BadgeHorizontalPosition, BadgeSize, BadgeVerticalPosition } from '../../../../src/directives/badge/index';
+
+@Component({
+    selector: 'app-badge',
+    templateUrl: './badge.testpage.component.html',
+    styleUrls: ['./badge.testpage.component.less'],
+})
+export class BadgeTestPageComponent {
+    sizes: BadgeSize[] = [ 'small', 'medium', 'large' ];
+    verticalPositions: BadgeVerticalPosition[]  = [ 'above', 'below' ];
+    horizontalPositions: BadgeHorizontalPosition[] = ['after', 'before'];
+    colorCombinations: Map<string, string> = new Map<string, string>([
+        [ 'warning', 'critical' ],
+        [ 'grey5', 'grey1' ],
+        [ 'chart1', 'chart2' ],
+        [ 'chart3', 'chart4' ]
```

Comment:
As discussed earlier, maybe hex and rgb values in here as well?

Reply:
I added a check to the karma tests. I will add another couple of examples of both in here too.